### PR TITLE
move user service to common

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[OIDC-Login für den Manager](https://trello.com/c/SRvq2hKs/1756-oidc-login-f%C3%BCr-den-manager)

**Comment by michaelmattig** [comment]
[OIDC-Login für den Manager](https://trello.com/c/3RU8ex1m/1782-oidc-login-f%C3%BCr-den-manager)

File: projects/common/src/lib/user/user.service.ts [hunk 1]
```diff
@@ -1,66 +1,215 @@
...
-    getSessionTokenForRequest(): Observable<string> {
-        return this.getSessionTokenStream().pipe(first());
+    getBackendStatus(): Observable<BackendStatus> {
+        return this.backendStatus$;
+    }
+
+    sessionFromBrowserOrCreateGuest(): Observable<Session> {
+        return this.restoreSessionFromBrowser().pipe(
+            catchError((error) => {
+                console.error(error);
```
Comment (ChristianBeilschmidt) [inline projects/common/src/lib/user/user.service.ts:154]: we always get the nasty error here:
```
ResponseError: Response returned an error code
    ResponseError runtime.js:258
    request runtime.js:193
    fulfilled runtime.js:23
    Angular 31
    adopt runtime.js:17
    ZoneAwarePromise Angular
    adopt runtime.js:16
    step runtime.js:36
    __awaiter runtime.js:38
    ZoneAwarePromise Angular
    __awaiter runtime.js:20
    fetchApi runtime.js:101
    request runtime.js:189
    fulfilled runtime.js:23
    Angular 12
[user.service.ts:154:24](http://localhost:4200/projects/common/src/lib/user/user.service.ts)
``

File: projects/common/src/lib/login/login.component.html [hunk 2]
```diff
@@ -61,7 +61,7 @@
                         <a routerLink="/register" *ngIf="canRegister">Register</a>
                     </ng-template>
                     <ng-template [ngSwitchCase]="FormStatus.LoggedIn">
-                        <button mat-raised-button (click)="$event.preventDefault(); redirectToMainView()" fxFlex>To Map</button>
+                        <button mat-raised-button (click)="$event.preventDefault(); redirectToMainView()" fxFlex>To App</button>
                         <button mat-raised-button (click)="$event.preventDefault(); logout()" fxFlex color="primary">Logout</button>
                     </ng-template>
```
Comment (ChristianBeilschmidt) [inline projects/common/src/lib/login/login.component.html:66]: when OIDC, how do I login as an admin?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/common/src/lib/login/login.component.html [hunk 3]
```diff
@@ -61,7 +61,7 @@
...
                 </mat-card-actions>
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/common/src/lib/login/login.component.html:67]: a session is a session is a session... I mean anonymous users can access everything via API, so why not let them use the manager?

I could implement some extra logic to auto logout guest users, but what do we gain from that?

File: projects/common/src/lib/user/user.service.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/common/src/lib/user/user.service.ts]: d436c32aa6b954a1e1b0715e343165c6908bb56d

File: projects/common/src/lib/login/login.component.html (hunk 2 above)
Comment (michaelmattig) [reply projects/common/src/lib/login/login.component.html:69]: c0ffd2d17a2a3bbb58b10e23a1e3b3a356049d96

File: projects/common/src/lib/user/user.service.ts [hunk 4]
```diff
@@ -1,66 +1,215 @@
...
+                    });
+                }
+            }
+            if (!status.available && this.sessionInitialized) {
+                this.sessionInitialized = false;
                 this.session$.next(undefined);
-            });
+                this.notificationService.error('Session close caused by backend shutdown');
+            }
+        });
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/common/src/lib/user/user.service.ts]: a6c38f18a73c34eb3d31a2d5818619090e699f06

please check if it all still works, risky refactor

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**